lib/ask_web/controllers/survey_controller.ex

```diff
@@ -58,6 +59,28 @@ defmodule AskWeb.SurveyController do
     render(conn, "index.json", surveys: surveys)
   end
 
+  def list_unused(conn, %{"project_id" => project_id}) do
+    project = load_project(conn, project_id)
+
+    surveys =
+      Repo.all(
+        from s in Survey,
+          left_join: r in Respondent,
+          on: r.survey_id == s.id,
+          where: s.project_id == ^project.id and s.state == :terminated,
+          select: %{survey_id: s.id, name: s.name, ended_at: s.ended_at, respondents: sum(fragment("if(?, ?, ?)", r.disposition == :registered, 1, 0))},
```

Comment:
I suppose this could also be a `select ..., count(*) as respondents from ... where r.disposition = 'registered'` which may be more efficient. I'm not sure about the Ecto syntax though.

Reply:
The Ecto syntax is not a problem:

```
        from s in Survey,
          left_join: r in Respondent,
          on: r.survey_id == s.id,
          where: s.project_id == ^project.id and s.state == :terminated and r.disposition == :registered,
          select: %{survey_id: s.id, name: s.name, ended_at: s.ended_at, respondents: count(s.id)},
          group_by: [s.id],
          order_by: [desc: count(s.id)]
```

But then you don't get the surveys that have 0 available respondents listed. I'd rather explicitely show that there are no available respondents rather than ignore the survey at all.

I could, though, add a comment on the code explaining this.

There's probably another query that can be written for that, but not sure it'd be better.

Reply:
Good point. You can use a `left join` for that, and `count(r.id)` or any other field that is `NULL` when joining an empty set.